app/app.js

```diff
@@ -350,6 +351,27 @@ app.post('/snap', [
           // Set CSS Media
           await page.emulateMedia(fnMedia);
 
+          // Compile cookies. We have to manually specify some extra info such
+          // as host/path in order to create a valid cookie.
+          let cookies = [];
+          fnCookies.split('; ').map((cookie) => {
+            let thisCookie = {};
+            const [name, value] = cookie.split('=');
+
+            thisCookie.url = fnUrl;
+            thisCookie.name = name;
+            thisCookie.value = value;
+
+            cookies.push(thisCookie);
+          });
+
+          // Set cookies.
+          cookies.forEach(async function(cookie) {
+            await page.setCookie(cookie).catch((err) => {
+              log.error(err);
```

Comment:
Yay for logging.

Reply:
![logging](https://user-images.githubusercontent.com/254753/50101578-9aa42580-01e8-11e9-80f5-6ff24ecd78e0.gif)